### PR TITLE
NXP-27225: merge dirty properties received on suggestion op

### PR DESCRIPTION
**Pull request by nmpcunha**
Disclaimer:
- Not sure if I have the correct formatter or not, but some of the code got reformatted. Please let me know if which formatter should I use;
- The second commit was an attempt to test the properties merge, which lead me to the need of using Mocks and ArgumentCaptors. In theory, by using that kind of approach the test will be unitary and not a kind of integration test. Still, please let me know if you have other approach for ensuring that the new code is properly tested;

Thanks!

### CODE REVIEW

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java [hunk 1]
```diff
@@ -77,27 +80,40 @@
...
+    @OperationMethod
+    public Blob run(DocumentRef docRef) {
+        DocumentModel docModel = coreSession.getDocument(docRef);
+        return run(docModel);
+    }
+
     @OperationMethod
     public Blob run(DocumentModel doc) {
+        if (updatedDocument != null) {
+            Arrays.asList(doc.getSchemas())
```
Comment (andreinechaev) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: please no nested loops:

```
Arrays.stream(doc.getSchemas())
                            .flatMap(s1 -> udoc.getProperties(s1).entrySet().stream())
                            .forEach(entry -> doc.setPropertyValue(entry.getKey(), (Serializable) entry.getValue()));
```

should do the job

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java [hunk 2]
```diff
@@ -77,27 +80,40 @@
...
+                  .forEach(schema -> updatedDocument.getProperties(
+                          schema).forEach((key, value) -> doc.setPropertyValue(key, (Serializable) value)));
+        }
 
-        List<EnrichmentMetadata> suggestions;
-        if (doc == null || (suggestions = modelServingService.predict(doc)) == null || suggestions.isEmpty()) {
```
Comment (covolution) [inline nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: Not sure why you changed this?  The current style was suggested in a code review

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java (hunk 1 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: Personally, in this example, I think a standard for loop will be much clearer

Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: depends, if nested loops don't scare you, then yes :)

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java (hunk 2 above)
Comment (andreinechaev) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: cleaner, I'd assume

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java (hunk 1 above)
Comment (nmpcunha) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: Thanks for your feedback!
Not afraid by nested loops (only two in fact), but I can understand your point @andreinechaev about the usage of streams here. I'll perform the change

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java (hunk 2 above)
Comment (nmpcunha) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: Two reasons:
- Initially for my mind map and for debugging;
- Then I found cleaner, more readable;

But I can change it to the original one.

**Comment by nmpcunha** [comment]
Regarding your question about real end to end test, the answer is no.
In theory, unit tests shouldn't rely on real services. In this specific case the change will update some properties on the document that is sent to the prediction service. So, that should be what we test, no? And that's why I created other class in order to have something more concrete to show and discuss.
How would you suggest to test this specific change, @covolution ?

File: nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java (hunk 2 above)
Comment (covolution) [reply nuxeo-ai-model/src/main/java/org/nuxeo/ai/model/serving/SuggestionOp.java]: Leave it, how you did it.

**Pull request merged**